File: DataStructure/118. Pascal's Triangle.py
# Given an integer numRows, return the first numRows of Pascal's triangle.

# In Pascal's triangle, each number is the sum of the two numbers directly above it as shown:

numRows = 5
# Output: [[1],[1,1],[1,2,1],[1,3,3,1],[1,4,6,4,1]]

class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        result = []
        
        for i in range(numRows):
            if i == 0:
                result.append([1])
                
            elif i == 1:
                result.append([1, 1])
            
            else :
                temp = []
                before = result[i - 1]
                
                for index in range(len(before) - 1 ):
                    add_sum = before[index] + before[index + 1]
                    temp.append(add_sum)
            
                result.append([1] + temp + [1])
        
        return result

File: DataStructure/118. Pascal's Triangle.cs
public class Solution {
    public IList<IList<int>> Generate(int numRows) {
        List<int>[] results = new List<int>[numRows];
        results[0] = new List<int>();
        results[0].Add(1);

        for (int row = 1; row <= numRows - 1; row++)
        {
            results[row] = new List<int>();
            results[row].Add(1);

            if (row > 1)
                for (int i = 1; i <= row - 1; i++)
                    results[row].Add(results[row - 1][i - 1] + results[row - 1][i]);

            results[row].Add(1);
        }

        return results;
    }
}

    